juniper_official/lldp/lldp-session-state.rule

```diff
@@ -0,0 +1,131 @@
+/*
+ * Detects LLDP (session) state changes and notifies when anomalies are found.
+ *
+ * Requirement 455 - https://paragon-automation.atlassian.net/browse/REQ-455
+ *
+ */
+healthbot {
+    topic lldp {
+        description "Monitors lldp information i.e. neighbor state and notify anomalies";
+        synopsis "LLDP session statistics analyzer";
+        rule check-lldp-session {
+            description "Collects lldp session state periodically and notify anomaly when state is down";
+            synopsis "LLDP session state analyzer";
+            /*
+             * Monitors the LLDP session state. color is set to red when state is down.
+             * Otherwise the color is set to green.
+             * 
+             * Use interface name as key for rule.
+             */
+            keys interface-name;
+            /*
+             * Sensor configuration to get data from network devices
+             */
+            sensor lldp-sensor {
+                synopsis "LLDP open-config sensor definition";
+                description "/lldp open-config sensor to collect telemetry data from network device";
+                open-config {
+                    sensor-name /lldp/;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined using sensor path. Map the longer sensor names
+             * to the shorter field names used in the rules.
+             */
+            field if-lldp-state {
+                sensor lldp-sensor {
+                    path /lldp/interfaces/interface/state/enabled;
+                }
+                type string;
+                description "Local interface LLDP state";
+            }
+            field interface-name {
+                sensor lldp-sensor {
+                    where "/lldp/interfaces/interface/@name =~ /{{interface-name}}/";
+                    path "/lldp/interfaces/interface/@name";
+                }
+                type string;
+                description "Interfaces to be monitored";
+            }
+            field neighbor-state {
+                reference {
+                    path "/topic[topic-name='lldp']/rule[rule-name='get-lldp-state']/field[interface-name='$interface-name']/neighbor-capability-state";
```

Comment:
references are not supported in EPIC. Only network rules can have references. 

Since the keys are same in both rules and is being derived from same sensor, why can't these rules be merged into single rule?

Reply:
Merged